itest/test_manager.go

```diff
@@ -340,23 +340,31 @@ func (tm *TestManager) WaitForFpVoteCast(t *testing.T, fpIns *service.FinalityPr
 	return lastVotedHeight
 }
 
-func (tm *TestManager) WaitForNFinalizedBlocks(t *testing.T, n int) []*types.BlockInfo {
+func (tm *TestManager) WaitForNFinalizedBlocksAndReturnTipHeight(t *testing.T, n uint) uint64 {
 	var (
-		blocks []*types.BlockInfo
-		err    error
+		firstFinalizedBlock *types.BlockInfo
+		err                 error
+		lastFinalizedBlock  *types.BlockInfo
 	)
+
 	require.Eventually(t, func() bool {
-		blocks, err = tm.BBNConsumerClient.QueryLatestFinalizedBlocks(uint64(n))
+		lastFinalizedBlock, err = tm.BBNConsumerClient.QueryLatestFinalizedBlock()
 		if err != nil {
 			t.Logf("failed to get the latest finalized block: %s", err.Error())
 			return false
 		}
-		return len(blocks) == n
+		if lastFinalizedBlock == nil {
+			return false
+		}
+		if firstFinalizedBlock == nil {
+			firstFinalizedBlock = lastFinalizedBlock
+		}
+		return lastFinalizedBlock.Height-firstFinalizedBlock.Height == uint64(n-1)
```

Comment:
```suggestion
		return lastFinalizedBlock.Height-firstFinalizedBlock.Height >= uint64(n-1)
```

I think this is the culprit that CI fails